test/new-e2e/test-infra-definition/cross_platform_test.go

```diff
@@ -0,0 +1,81 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package testinfradefinition
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/DataDog/datadog-agent/test/new-e2e/utils/e2e"
+	"github.com/DataDog/test-infra-definitions/components/datadog/agentparams"
+	"github.com/DataDog/test-infra-definitions/scenarios/aws/vm/ec2os"
+	"github.com/DataDog/test-infra-definitions/scenarios/aws/vm/ec2params"
+	"github.com/stretchr/testify/require"
+)
+
+type crossPlatformSuite struct {
+	e2e.Suite[e2e.AgentEnv]
+}
+
+func TestCrossPlatformSuite(t *testing.T) {
+	e2e.Run(t, &crossPlatformSuite{}, e2e.AgentStackDef(nil))
+}
+
+func (s *crossPlatformSuite) TestUbuntuOS() {
+	os := ec2os.UbuntuOS
+	s.subTestInstallAgentVersion(os)
+	s.subTestInstalAgent(os)
+}
+
+func (s *crossPlatformSuite) TestWindows() {
+	// As the Windows installer cannot downgrade an agent version,
+	// the order of tests matter. New tests must be added at
+	// the end of the function.
+	os := ec2os.WindowsOS
+	s.subTestInstallAgentVersion(os)
+	s.subTestInstalAgent(os)
+}
+
+func (s *crossPlatformSuite) TestOtherOSES() {
+	for _, os := range ec2os.GetOSTypes() {
+		if os == ec2os.WindowsOS || os == ec2os.UbuntuOS {
+			// Windows and Ubuntu have their dedicated tests
+			continue
+		}
+
+		if os == ec2os.CentOS || os == ec2os.RockyLinux {
+			// TODO: There is a bug for CentOS and RockyLinux
+			continue
+		}
+
+		s.subTestInstalAgent(os)
+	}
+}
+
+func (s *crossPlatformSuite) subTestInstalAgent(os ec2os.Type) {
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement. -->
**🥜 nitpick**
Typo in the name
```suggestion
func (s *crossPlatformSuite) subTestInstallAgent(os ec2os.Type) {
```

Reply:
Code updated